src/utils/hooks/useOutcomes.tsx

```diff
@@ -0,0 +1,122 @@
+import { useEffect, useState } from 'react'
+import { getRequest } from '../../utils/helpers/fetchMethods'
+import config from '../../config'
+import { LOCAL_STORAGE_JWT_KEY } from '../data/globalConstants'
+import { TemplatePermissions } from '../types'
+import {
+  OutcomesResponse,
+  OutcomesTableResponse,
+  OutcomesDetailResponse,
+  OutcomeTableAPIQueries,
+} from '../../utils/types'
+const serverURL = config.serverREST
+
+// 3 simple hooks for returning Outcome state
+
+interface OutcomeListProps {
+  templatePermissions?: TemplatePermissions
+}
+
+interface OutcomeTableProps extends OutcomeListProps {
+  tableName: string
+  apiQueries: OutcomeTableAPIQueries
+}
+
+interface OutcomeDetailsProps extends OutcomeListProps {
+  tableName: string
+  recordId: number
+}
+
+type ErrorResponse = {
+  statusCode: number
+  error: string
+  message: string
+}
+
+export const useOutcomesList = ({ templatePermissions }: OutcomeListProps) => {
+  // Note: we don't *need* templatePermissions, we only pass it in so that the
+  // hook can react to changes, since JWT (what we *actually* need) is not in
+  // State (perhaps it should be?)
```

Comment:
Not sure what you mean by "not in state" I think this is included in userState. Maybe you can use this here?

Reply:
I just mean that nothing changes in response to the JWT changing, as it's not stored anywhere in UserState. So we can't pass it in to the dependency array here. So instead I'm just passing in "templatePermissions", which is derived from JWT, and *is* in UserState, so we can react to it changing.

Reply:
Ok - I might not be seeing exactly what you would like to change this to be, so you are considering that the UsetState not only store the templatePermissions but also the JWT received. This is because the templatePermissions don't change when we get a new JWT? I'm not really following this part...
Also, you don't access UserState in the `useOutcomes` hook so it's not clear what change you were thinking. Maybe a code example here might help... What I'm understanding is - instead of passing `templatePermissions` as props you could just access it from UserState and trigger `useEffect` based on that. For example:
```
export const useOutcomesList = () => {
  // Note: we don't *need* templatePermissions, we only pass it in so that the
  // hook can react to changes, since JWT (what we *actually* need) is not in
  // State (perhaps it should be?)
  const [error, setError] = useState<ErrorResponse | null>(null)
  const [loading, setLoading] = useState(false)
  const [outcomesList, setOutcomesList] = useState<OutcomesResponse>([])
  const {
    userState: { templatePermissions },
  } = useUserState()

  useEffect(() => {
    const JWT = localStorage.getItem(LOCAL_STORAGE_JWT_KEY)
    if (!JWT) return
    const url = `${serverURL}/outcomes`
    processRequest(url, JWT, setError, setLoading, setOutcomesList)
  }, [templatePermissions])

  return { error, loading, outcomesList }
}
```

But this can be done already, no?

Reply:
Yes, okay, I think it makes sense to get templatePermissions from UserState rather than passing it in. Thanks, I'll make this change shortly.

Reply:
Cool, this is much nicer now, thanks for the suggestion 2d4620a0

(Change pushed to later branch though)